feat : update studio docs (#675)
agno-agi/docs: agent-os/studio/introduction.mdx

description: "A visual editor in AgentOS to build Agents, Teams, and Workflows."
---

Build and orchestrate Agents, Teams, and Workflows on a live canvas to deploy production-ready agentic systems with AgentOS Studio.

## Concepts

> **[Agents](/agent-os/studio/agents)**

Reviewer: ```suggestion
**[Agents](/agent-os/studio/agents)** :
```

Build agent by giving it a model, tools, and instructions. Add knowledge and memory to ground its responses and remember context.

**[Teams](/agent-os/studio/teams)**
Build multi-agents team that works toward a shared goal. Choose how the leader coordinates with members using `coordinate`, `route`, `broadcast`, or `tasks` mode.

**[Workflows](/agent-os/studio/workflows)**
Orchestrate agents and teams into step-based pipelines. Control the flow with loops, conditions, routers, and parallel execution.

> use functions or cel expressions to evaluate conditions and selectors.

Reviewer: Does this belong to workflows only or agent and teams too ? looks like an orphaned line

**[Registry](/agent-os/studio/registry)**
Add tools, models, databases, and schemas that Studio can use to build agents, teams, and workflows.

Register agents and teams in the `Registry` to reuse them as members in Studio teams and as steps in Studio workflows.

> Register `knowledge`, `memory_managers`, and `session_summary_managers` for agents and teams to use.

Reviewer: ```suggestion
Register the components Studio composes with: tools, models, databases, schemas, knowledge, `memory_managers`, and `session_summary_managers`. Register existing agents and teams to reuse them as members in Studio teams or steps in Studio workflows.
```

### 1. Build

Create your agent, team, or workflow using the visual builder. use tools, models, and knowledge bases from the Registry, add instructions and configure the settings.

  <img
    src="/images/agent-os-studio-create.png"
    alt="Create an agent in Studio"
  />

> ### 2. Save Draft or publish directly

Reviewer: ```suggestion
### 2. Save Draft or Publish Directly
```

- Save your work as a draft or publish directly. Drafts can be edited and updated. Drafts helps you test your component before publishing it.
- Publish or Draft multiple versions to checkpoint your progress.
  <img
    src="/images/agent-os-studio-draft-publish-btn.png"
    alt="Save draft or publish directly in Studio"
    style={{ borderRadius: "8px" }}

>   />

Reviewer: We don't need to show this button image, it doesnot give any idea . we can skip an image here

### 3. Test

Once saved, test your draft or published version in the AgentOS:

- **Chat Page**: 
  - Interact with your agent, team, or workflow in real-time

>   - run specific version of the component by selecting it in the dropdown.

Reviewer: ```suggestion
  - Run specific version of the component by selecting it in the dropdown.
```

  <img
    src="/images/agent-os-studio-chat-page-versioning.png"
    alt="Run specific version of the component in the chat page"
    style={{ borderRadius: "8px" }}
  />      

- **View Traces**: Inspect tool calls, model responses, and reasoning for each run
- **Debug Mode**: Enable verbose logging to troubleshoot issues

<Warning>
  Before publishing, test and make sure your agent handles edge cases and
  unexpected inputs gracefully.
</Warning>

> ### 4. Delete

Reviewer: Don't think we should include delete as a seperate point, it  can be a part inside Manage versions? or skip it entirely

- Delete the component from the Studio.

### 5. Manage Versions

Access the full version history for any agent, team, or workflow:

<img
  src="/images/agent-os-studio-versions-page.png"
  alt="Versions page in Studio"
  style={{ borderRadius: "16px" }}
/>
- **Restore**: edit and update the draft version.
- **Set Current**: Choose which published version is used by default when running via the API

- only draft versions can be updated and edited in the Studio.
- published versions are immutable and can only be updated by publishing a new version.
- draft version cannot be set as current.

> - only draft versions can be deleted from versions page , published versions and current config cannot be deleted.

Reviewer: Lets start each sentence with a capital character

Reviewer: ```suggestion
Two actions are available on this page:

- **Restore**: open a draft for editing in Studio.
- **Set Current**: choose which published version the API serves.

Drafts are mutable: you can edit them in Studio and delete them. Published versions are immutable and undeletable. To change a published version, publish a new one. Only published versions can be set as current.
```
